src/phpDocumentor/Configuration/ConfigurationFactory.php

```diff
@@ -0,0 +1,332 @@
+        $ignoreSymlinks     = true;
+
+        if (isset($phpDocumentor->parser)) {
+            $extensions = $this->buildExtensionsPart($phpDocumentor->parser);
+            $markers    = $this->buildMarkersPart($phpDocumentor->parser);
+
+            $visibility         = ((string) $phpDocumentor->parser->visibility) ?: $visibility;
+            $defaultPackageName = ((string) $phpDocumentor->parser->{'default-package-name'}) ?: $defaultPackageName;
+            $template           = ((string) $phpDocumentor->transformations->template->attributes()->name) ?: $template;
+
+            if (isset($phpDocumentor->parser->files)) {
+                if (isset($phpDocumentor->parser->files->{'ignore-hidden'})) {
+                    $ignoreHidden = filter_var($phpDocumentor->parser->files->{'ignore-hidden'}, FILTER_VALIDATE_BOOLEAN);
+                }
+
+                if (isset($phpDocumentor->parser->files->{'ignore-symlinks'})) {
+                    $ignoreSymlinks = filter_var($phpDocumentor->parser->files->{'ignore-symlinks'}, FILTER_VALIDATE_BOOLEAN);
+                }
+            }
+        }
+
+        $phpdoc2Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => 'file://build/docs',
+                    'cache'  => '/tmp/phpdoc-doc-cache'
+                ],
+                'versions'  => [
+                    '1.0.0' => [
+                        'folder' => 'latest',
+                        'api'    => [
+                            'format'               => 'php',
+                            'source'               => [
+                                'dsn'   => 'file://.',
+                                'paths' => [
+                                    0 => 'src'
+                                ]
+                            ],
+                            'ignore'               => [
+                                'hidden'   => $ignoreHidden,
+                                'symlinks' => $ignoreSymlinks,
+                                'paths'    => [
+                                    0 => 'src/ServiceDefinitions.php'
+                                ]
+                            ],
+                            'extensions'           => $extensions,
+                            'visibility'           => $visibility,
+                            'default-package-name' => $defaultPackageName,
+                            'markers'              => $markers,
+                        ],
+                        'guide'  => [
+                            'format' => 'rst',
+                            'source' => [
+                                'dsn'   => 'file://../phpDocumentor/phpDocumentor2',
+                                'paths' => [
+                                    0 => 'docs'
+                                ]
+                            ]
+                        ]
+                    ]
+                ],
+                'templates' => [
```

Comment:
There may be multiple `<template>` tags in the phpDocumentor2 XML because you can execute multiple templates in succession